Revamp README

felipecrs opened the pull request:
For the first time ever, I was able to control my Tuya locks from within Home Assistant.

I wanted to contribute back somehow, and I noticed the README could use some improvements.

I have also ensured the integration is called _Xtend Tuya_ everywhere, for consistency.

I also formatted `README.md` with prettier.

If you like it, I can send an additional PR reviewing the other documents.

felipecrs commented:
Reason to remove the manual installation steps:

People who prefer not to use HACS are very advanced ones. They will know how to install an integration when not using HACS, no need to tell them these steps.

felipecrs commented:
Also, I've gotta say the _Supported operation modes_ is completely confusing to me. Maybe we can improve it in the future. I've moved it to the bottom of the page for now, since it is not required for the person to understand it to leverage this integration.

azerty9971 commented:
My first ever PR <3

Thank you so much for this, you made my day!!!

The supported operation modes is there to tell people what each mode has for pros/cons...
But I'm considering more and more to recommend people to simply use XT without the regular Tuya integration since this causes bugs that are hard to track and Tuya's evolution might break XT.

I think it's better to keep only the section about using XT as a replacement of Tuya (so either disable the Tuya integration or disable the config entries)

Again, thank you so much!!
Azerty

felipecrs commented:
Very interesting. I didn't know I could remove the official Tuya integration.

azerty9971 commented:
Yes, basically, if you do then XT fetches the DPCodes from the regular Tuya integration and handles them :)

felipecrs commented:
@azerty9971 with this new information, now I understand the table. And I refactored it too, and moved it back to the top.

Preview: https://github.com/felipecrs/xtend_tuya/blob/revamp-readme/README.md

As per removing support for official Tuya + Xtend, I suggest you do that in a separate PR. Personally, I like support to work alongside the official Tuya integration as it doesn't require me to refactor my dashboards/automations for the new entity ids or reapply the customizations in the entity names or ids that I had added to the official integration during the past years.

azerty9971 commented:
I love the new README, thanks!!!

You're right about the support for official tuya + XT, don't hesitate to open bug reports for anything strange you see in this mode as I moved my prod HA to standalone XT (only my dev system has it configured)

Awesome work!!

felipecrs commented:
Still, I think it's your call whether to support running alongside the official Tuya integration or not.

For example, I agree your time would be better invested into adding new features to the integration rather than making it work alongside the existing integration.

For what it is worth, I support whatever decision you make.
